<!DOCTYPE HTML>
<html lang="zh-CN">
	<head>
		<meta charset="GB2312">
		<title>异常处理</title>
		<link rel="stylesheet" href="../images/css/book.css">
		<link rel="stylesheet" href="../images/css/prettify.css">
		<script type="text/javascript" src="../images/js/jquery-1.7.1.min.js"></script>
		<script type="text/javascript" src="../images/js/prettify.js"></script>
		<script type="text/javascript">
			$(function(){	
				$(window).resize(function(){
					$('.book-content').css('min-height', $(window).height() - 130);
				}).resize();
				
				//表格隔行变色
				$('table').TableColor();
				
				//代码高亮
				prettyPrint();
			});
			
			//表格隔行变色插件
			$.fn.TableColor = function(){
				return $(this).each(function(){
					if(this.nodeName.toLowerCase() != 'table') return;
					var self = $(this);
					self.find('tr').each(function(index) {
						var _this = $(this);
						if(index % 2 == 0){
							_this.addClass('add');
						} else {
							_this.addClass('even');	
						}
						_this.hover(
							function(){_this.addClass('hover')},
							function(){_this.removeClass('hover')}
						);
					});	
				});
			}
		</script>
	</head>
	<body><script language="javascript">
		if ((window.name != "content") && (navigator.userAgent.indexOf("Opera") <= -1) )
		document.write("<center><a href='../index.html?page=html/127.html'>show framing</a></center>")
		</script>
		<div class="book-title">
			<h1>异常处理</h1>
			<div class="page"><a title="上一页" class="prev" href="126.html">上一页</a><a title="下一页" class="next" href="128.html">下一页</a></div>
		</div>
		<div class="book-content">和PHP默认的异常处理不同，ThinkPHP抛出的不是单纯的错误信息，而是一个人性化的错误页面，如下图所示：<br />
<img src="../images/img/3.png" /><br />
只有在调试模式下面才能显示具体的错误信息，如果在部署模式下面，你可能看到的是一个统一错误的提示文字，如果你试图在部署模式下访问一个不存在的模块或者操作，会发送404错误。<br />
调试模式下面一旦系统发生严重错误会自动抛出异常，也可以用ThinkPHP定义的throw_exception方法手动抛出异常。<table border="0" cellspacing="1" cellpadding="0">    <tr>        <th colspan="2">throw_exception  抛出异常</th>    </tr>    <tr>        <td>用法</td>        <td>throw_exception($msg, $type='ThinkException', $code=0)</td>    </tr>    <tr>        <td>参数</td>        <td>msg（必须）：要抛出的异常信息，字符串<br />
type（可选）：异常类型，默认为ThinkException，如果异常类型不存在，则会调用系统的halt方法直接输出错误信息。<br />
code（可选）：异常代码，默认为0<br />
        </td>    </tr>    <tr>        <td>返回值</td>        <td>无</td>    </tr></table>下面是throw_exception函数的一些使用例子：<code class="prettyprint linenums lang-php">throw_exception('新增失败');<br />
throw_exception('信息录入错误','InfoException');</code>同样也可以使用throw 关键字来抛出异常，下面的写法是等效的：<code class="prettyprint linenums lang-php">throw&nbsp;new&nbsp;ThinkException('新增失败');<br />
throw&nbsp;new&nbsp;InfoException('信息录入错误');</code>如果需要，我们建议在项目的类库目录下面增加Exception目录用于专门存放异常类库，以更加精确地定位异常。</div>
		<div class="book-footer">
			<div class="page"><a title="上一页" class="prev" href="126.html">上一页</a><a title="下一页" class="next" href="128.html">下一页</a></div>
		</div>
	</body>
</html>